engine/routes.py

```diff
@@ -159,6 +170,35 @@ def estimate():
     return jsonify(util.calc.get_estimate(input_data, request.url))
 
 
+@app.route('/stats', methods=['POST'])
+@set_timeout()
+def stats():
+    """
+    Return the calculated download size and completion time for a completed request by looking at the output directory
+    """
+    input_data = request.get_json()
+
+    try:
+        if app.config['AGGREGATE']:
+            async_job_dir = get_final_dir(input_data)
+        else:
+            async_job_dir = get_local_dir(input_data)
+
+        if os.path.isdir(async_job_dir):
+            return jsonify(util.calc.get_stats(async_job_dir))
+        else:
```

Comment:
Since the "if" leads to an exit from this function, no need for the "else".

Reply:
Removed extraneous else. Negated if statement and put the exception in the if statement instead of the return so that the return is at the end of the code block while the exception terminates it early.